src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -411,19 +411,25 @@ internal sealed override ObsoleteAttributeData ObsoleteAttributeData
                     return UnmanagedCallersOnlyAttributeData.Uninitialized;
                 }
 
+                var earlyData = (CommonMethodEarlyWellKnownAttributeData?)lazyCustomAttributesBag.EarlyDecodedWellKnownAttributeData;
                 if (lazyCustomAttributesBag.IsDecodedWellKnownAttributeDataComputed)
                 {
-                    var data = (MethodWellKnownAttributeData?)lazyCustomAttributesBag.DecodedWellKnownAttributeData;
+                    var lateData = (MethodWellKnownAttributeData?)lazyCustomAttributesBag.DecodedWellKnownAttributeData;
+                    if (earlyData is { UnmanagedCallersOnlyAttributePresent: true } && lateData is null or { UnmanagedCallersOnlyAttributeData: null })
```

Comment:
>earlyData [](start = 24, length = 9)

Is there a guarantee that early data are always computed when all attributes are decoded? It feels like this check for both early and late data is unnecessary complicated. Why can't we just rely on late data?

Reply:
>  Why can't we just rely on late data? 

If the attribute body is malformed (see `UnmanagedCallersOnly_BadExpressionInArguments`) for an example, we never attempt to bind the attribute at all. So the early data records the presence of an unmanaged callers only attribute, but the late data contains no such info. Now, we could not take that into account here, and the consequence would be that methods that convert such a method to a function pointer would be treated as having a default managed calling convention, as opposed to having a default unmanaged calling convention (which is what this code will do). I felt that the approach that would feel the best for users is to treat it as default unmanaged, rather than default managed, but I could revise it to return null in that case if you think it's too complicated.

---
In reply to: [488957958](https://github.com/dotnet/roslyn/pull/47651#discussion_r488957958) [](ancestors = 488957958)

Reply:
> I could revise it to return null in that case if you think it's too complicated.

I think this is too much trouble for little benefit. I think it is our usual approach to ignore malformed well-known attributes as if they were not applied. I think we should follow the same approach with this attribute.

---
In reply to: [490554943](https://github.com/dotnet/roslyn/pull/47651#discussion_r490554943) [](ancestors = 490554943,488957958)